File: OPENAPI-REST-API/swagger-client/scala-akka-http-server/src/main/scala/io/swagger/server/model/FloatingIpOrderRequest.scala
package io.swagger.server.model


/**
 * Request body to validate (putFloating_ips) or place (addFloatingIp) a floating IP order. Service type ids come from getNewFloatingIp.serviceTypes.
 *
 * @param serviceType Floating IP plan service type id from getNewFloatingIp.serviceTypes (must be buyable).
 * @param coupon Coupon code.
 * @param comment Free-form note saved on the service row (used on addFloatingIp).
 */
case class FloatingIpOrderRequest (
  serviceType: Int,
  coupon: Option[String],
  comment: Option[String]
)

File: OPENAPI-REST-API/openapi-client/javascript-closure-angular/API/Client/FloatingIpOrderRequest.js
goog.provide('API.Client.FloatingIpOrderRequest');

/**
 * Request body to validate (putFloating_ips) or place (addFloatingIp) a floating IP order. Service type ids come from getNewFloatingIp.serviceTypes.
 * @record
 */
API.Client.FloatingIpOrderRequest = function() {}

/**
 * Floating IP plan service type id from getNewFloatingIp.serviceTypes (must be buyable).
 * @type {!number}
 * @export
 */
API.Client.FloatingIpOrderRequest.prototype.serviceType;

/**
 * Coupon code.
 * @type {!string}
 * @export
 */
API.Client.FloatingIpOrderRequest.prototype.coupon;

/**
 * Free-form note saved on the service row (used on addFloatingIp).
 * @type {!string}
 * @export
 */
API.Client.FloatingIpOrderRequest.prototype.comment;

